common/src/main/kotlin/entity/Snowflake.kt

```diff
@@ -72,13 +98,33 @@ class Snowflake(val value: ULong) : Comparable<Snowflake> {
     }
 
     companion object {
-        private const val discordEpochLong = 1420070400000L                                 // 42 one bits
-        private const val maxMillisecondsSinceDiscordEpoch = 0b111111111111111111111111111111111111111111L
+        private const val nonTimestampBitCount = 22
+        private const val discordEpochLong = 1420070400000L
+
+        /**
+         * A range that contains all valid raw Snowflake [value]s.
+         *
+         * Note that this range might change in the future.
+         */
+        val validValues: ULongRange = ULong.MIN_VALUE..Long.MAX_VALUE.toULong() // 0..9223372036854775807
```

Comment:
Should I use literals here?
like: `val validValues: ULongRange = 0uL..9223372036854775807uL`

Reply:
it'd be better

Reply:
I disagree, magic numbers should be avoided as much as possible

Reply:
sounds like a fine argument to me :+1:

Reply:
And I have a comment for it anyway, so it should be fine